// react-native-maps/src/index.ts
import MapView, {
  AnimatedMapView as Animated,
  MAP_TYPES,
  type MapViewProps,
} from 'react-native-maps/src/MapView';

import Marker from 'react-native-maps/src/MapMarker';
export {MapMarker} from 'react-native-maps/src/MapMarker';
export type {MapMarkerProps} from 'react-native-maps/src/MapMarker';

import Overlay from 'react-native-maps/src/MapOverlay';
export {MapOverlay} from 'react-native-maps/src/MapOverlay';
export type {MapOverlayProps} from 'react-native-maps/src/MapOverlay';

export {default as Polyline, MapPolyline} from 'react-native-maps/src/MapPolyline';
export type {MapPolylineProps} from 'react-native-maps/src/MapPolyline';
export {default as Heatmap, MapHeatmap} from 'react-native-maps/src/MapHeatmap';
export type {MapHeatmapProps} from 'react-native-maps/src/MapHeatmap';
export {default as Polygon, MapPolygon} from 'react-native-maps/src/MapPolygon';
export type {MapPolygonProps} from 'react-native-maps/src/MapPolygon';
export {default as Circle, MapCircle} from 'react-native-maps/src/MapCircle';
export type {MapCircleProps} from 'react-native-maps/src/MapCircle';
export {default as UrlTile, MapUrlTile} from 'react-native-maps/src/MapUrlTile';
export type {MapUrlTileProps} from 'react-native-maps/src/MapUrlTile';
export {default as WMSTile, MapWMSTile} from 'react-native-maps/src/MapWMSTile';
export type {MapWMSTileProps} from 'react-native-maps/src/MapWMSTile';
export {default as LocalTile, MapLocalTile} from 'react-native-maps/src/MapLocalTile';
export type {MapLocalTileProps} from 'react-native-maps/src/MapLocalTile';
export {default as Callout, MapCallout} from 'react-native-maps/src/MapCallout';
export type {MapCalloutProps} from 'react-native-maps/src/MapCallout';
export {
  default as CalloutSubview,
  MapCalloutSubview,
} from 'react-native-maps/src/MapCalloutSubview';
export type {MapCalloutSubviewProps} from 'react-native-maps/src/MapCalloutSubview';
export {default as AnimatedRegion} from 'react-native-maps/src/AnimatedRegion';
export {default as Geojson} from 'react-native-maps/src/Geojson';
export type {GeojsonProps} from 'react-native-maps/src/Geojson';

export {Marker, Overlay};
export type {MapViewProps};
export {Animated, MAP_TYPES};

export * from 'react-native-maps/src/ProviderConstants';
export * from 'react-native-maps/src/MapView.types';
export * from 'react-native-maps/src/MapPolygon.types';
export * from 'react-native-maps/src/sharedTypes';

export const MarkerAnimated = Marker.Animated;
export const OverlayAnimated = Overlay.Animated;

export default MapView;
